habitica.el

```diff
@@ -343,6 +349,31 @@ Options are 0.1, 1, 1.5, 2; eqivalent of Trivial, Easy, Medium, Hard."
   "Feed PET using FOOD."
   (habitica--send-request (format "/user/feed/%s/%s" pet food) "POST" ""))
 
+(defun habitica-api-get-groups (&rest types)
+  "Get groups which type is in TYPES."
+  (let* ((types (or types '("party" "guilds" "privateGuilds" "publicGuilds" "tavern")))
+         (types (string-join types ",")))
+    (message "types=%s" types)
+    (habitica--send-request (format "/groups?type=%s" (url-encode-url types)) "GET" "")))
+
+(defun habitica-api-get-group-id (&optional name)
```

Comment:
what happens if the name is empty?

Reply:
```emacs-lisp
         (name (or name (completing-read "Select the group: "  (mapcar get-group-name-fn groups))))
```
if `name` is empty,  the function will list out all the group names and ask the user to select a group.